### ISSUE TITLE
Add ITestDataBuilder<TObject> interface, missing after migration from NTestDataBuilder

### ISSUE THREAD

**Issue by dennisroche** [TestStack/TestStack.Dossier#47]
I found this interface useful to create a extension method for seeding a DB context directly with builders.

e.g. 

```cs
public static async Task<T> SaveAsync<T>(this IDbContext context, ITestDataBuilder<T> builder)
    where T : class
{
    var entity = builder.Build();
    context.Set<T>().Add(entity);
    await context.SaveAsync();
    return entity;
}
```

It was in `NTestDataBuilder`, however was removed for some reason in `TestStack.Dossier`. The change wasn't listed in [BREAKING_CHANGES.md](https://github.com/TestStack/TestStack.Dossier/blob/master/BREAKING_CHANGES.md)

I'll raise a PR for this soon.

**Comment by cottsak** [comment]
It was probably removed because store-specific behaviour like this very likely doesn't have a place in a lib designed for the 'building' of objects primarily. How they're saved and loaded is kinda out of scope.

Can we answer this one @robdmoore ?

**Comment by robdmoore** [comment]
This wasn't a deliberate omission. I probably just didn't see the point in the interface when I reworked it in the addition of the v3 features. I found the commit where it happened.

I'm happy enough with this change. How they are saved and loaded is out of scope for Dossier, but I certainly don't mind if people have the flexibility to add that on themselves when using it (I've done similar myself - it's very handy).

I'll merge and release.

Thanks!
